src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -217,17 +220,17 @@ const FiatDetailsScreen = ({ route, navigation }: Props) => {
         <SafeAreaView style={styles.container} edges={['bottom']}>
           <KeyboardAwareScrollView contentContainerStyle={styles.contentContainers}>
             <View>
-              {formFields.map((field, index) => (
+              {formFields.map((field) => (
                 <FormField
-                  key={`${field}-${index}`}
+                  key={field.name}
                   field={field}
-                  index={index}
-                  value={fieldValues.current[index]}
-                  errorMessage={errors.get(index)}
-                  onChange={(value) => {
-                    setInputValue(value, index)
+                  value={fieldNamesToValues.current[field.name]}
+                  errorMessage={errors.get(field.name)}
+                  onChange={(fieldName, value) => {
+                    setInputValue(fieldName, value)
                   }}
                   allowedValues={quote.getFiatAccountSchemaAllowedValues(field.name)}
+                  fieldNamesToValues={fieldNamesToValues.current}
```

Comment:
Would it be more straightforward to just have a `isVisible` prop on FormField?

Reply:
➕